src/hooks/useLocations.js

```diff
@@ -15,30 +16,48 @@ export const useLocations = (lat, lng) => {
     const baseURL = 'https://segdeha.com/api/nearby.php';
     const completeURL = lat && lng && `${baseURL}?lat=${lat}&lng=${lng}`;
 
-    setStatus('loading');
-    fetch(completeURL, { signal })
-      .then((response) => {
-        setIsFetching(true);
-        return response.json();
-      })
-      .then((data) => {
-        setLocations(data?.query?.pages);
-        setStatus('success');
-      })
-      .catch((err) => {
-        if (err.name !== 'AbortError') {
-          setStatus('error');
-          setError(err);
-        }
-      })
-      .finally(() => setIsFetching(false));
-
-    // Clean up
-    return function cancel() {
-      setStatus('cancelled');
-      abortController?.abort();
+    const fetchLocations = () => {
+      setStatus('loading');
+      fetch(completeURL, { signal })
+        .then((response) => {
+          setIsFetching(true);
+          return response.json();
+        })
+        .then((data) => {
+          setLocations(data?.query?.pages);
+          setStatus('success');
+        })
+        .catch((err) => {
+          if (err.name !== 'AbortError') {
+            setStatus('error');
+            setError(err);
+          }
+        })
+        .finally(() => setIsFetching(false));
+
+      //Clean up
+      return function cancel() {
+        setStatus('cancelled');
+        abortController?.abort();
+      };
     };
-  }, [lat, lng]);
+
+    if (isDragged) {
+      fetchLocations();
+      return;
+    }
+
+    if (Date.now() >= timeRef.current + interval) {
+      timeRef.current = Date.now();
+      fetchLocations();
+    } else {
+      const id = setTimeout(() => {
+        timeRef.current = Date.now();
+        fetchLocations();
+      }, interval - (Date.now() - timeRef.current));
+      return () => clearTimeout(id);
```

Comment:
I don’t _think_ you need to return the result of the `clearTimeout` call here, but correct me if I’m wrong. One kind of code smell here is that you only return from the `else` case and not from the `if` case. Earlier in the method, you return no value (which returns `undefined`), which is a totally valid way to shortcut a method’s execution, but my guess is nothing is consuming the return value, right?

Reply:
@segdeha, it's being returned here so that we can tidy up the last useEffect and we don’t get a weird leak where the useEffect will stack

Reply:
OK, cool. I'm happy as long as we're consistent about returning a value in each case. Thank you!